### PR TITLE
[IA-958][risk=no]Add Safe Mode extension to repo

### PR DESCRIPTION
**Pull request by jmthibault79**
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/jupyter/safe-mode.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        .append(
+            '<style>' +
+              '#notification_notebook { display: none; } ' +
+              '#safe-mode { background-color: #FFFFB2; }' +
+              '</style>');
+
+    // Add our own persistent "Safe Mode" notification next to the other
+    // notifications, e.g. kernel status.
+
+    // TODO: convert to tooltip with this text:
```
Comment (jmthibault79) [inline src/main/resources/jupyter/safe-mode.js]: Final UI will be tracked in a separate story.

File: src/main/resources/jupyter/safe-mode.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    // "notbook" is an intentional typo to match the Jupyter UI HTML.
+    $('#save-notbook').remove();
+    $('#save_notebook_as').remove();
+    $('#save_checkpoint').remove();
+
+    $('#notification_area')
+        .append(
+            '<style>' +
+              '#notification_notebook { display: none; } ' +
```
Comment (jmthibault79) [inline src/main/resources/jupyter/safe-mode.js]: This is not yet working as intended, so I'm investigating: I still see "Notebook saved" messages sometimes.

Comment (jmthibault79) [reply src/main/resources/jupyter/safe-mode.js]: I think my "span" commit fixed it.  Testing now...

Comment (jmthibault79) [reply src/main/resources/jupyter/safe-mode.js]: yep!

**Comment by jmthibault79** [comment]
Closing in favor of #945, which includes this.

File: src/main/resources/jupyter/safe-mode.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    'base/js/namespace'
+], (Jupyter) => {
+  const load = () => {
+    // TODO: query welder for when to enable (IA-979).   always-on, for now
+    const enabled = true;
+    if (!enabled) {
+      return;
+    }
+
+    // Disable UI controls/notifications relating to saving.
```
Comment (calbach) [inline src/main/resources/jupyter/safe-mode.js]: We should disable the "new notebook" menu item and button for now as well. Marty's recommendation is that "create a new notebook" should take you back to the linked source directory; we're not quite ready for that so we can just hide it for now.

**Comment by calbach** [comment]
Sorry, had buffered comments on the wrong PR. Ignore these, will review your new PR.

File: src/main/resources/jupyter/safe-mode.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+// Adapted from the All of Us Researcher Workbench "Playground Mode"
+// https://github.com/all-of-us/workbench/blob/master/api/cluster-resources/playground-extension.js
+
+// TODO: the following is required until IA-979 is implemented
+// To deploy on a Leonardo cluster:
+// 1. copy to a public GCS location
+// 2. issue a PUT request to <leonardo>/api/cluster/<billing project>/<cluster name>
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply src/main/resources/jupyter/safe-mode.js]: This is the "create cluster" endpoint.  I hoped that we could use Update Cluster but that only handles VM sizing

File: src/main/resources/jupyter/safe-mode.js (hunk 3 above)
Comment (jmthibault79) [reply src/main/resources/jupyter/safe-mode.js]: Next PR does that :)

**Pull request closed without merging**